api/database/seeders/DatabaseSeeder.php

```diff
@@ -87,16 +87,60 @@ public function run()
         $pool = Pool::whereNotNull('published_at')->inRandomOrder()->first();
         $this->seedPoolCandidate($applicant, $pool);
         $this->seedAwardExperienceForPool($applicant, $digitalTalentPool);
-        $applicantUserSkills = $applicant->userSkills;
+        $applicantUserSkills = $applicant->userSkills->load(['skill']);
+        $applicantUserSkillTechnicalIndex = 0;
+        $applicantUserSkillBehaviouralIndex = 0;
         foreach ($applicantUserSkills as $applicantUserSkill) {
-            $isSkillLevelSet = $faker->boolean(75);
-            $isWhenSkillUsedSet = $faker->boolean(75);
-            if ($isSkillLevelSet) {
+            if ($faker->boolean(75)) {
                 $applicantUserSkill->skill_level = $faker->randomElement(ApiEnums::skillLevels());
             }
-            if ($isWhenSkillUsedSet) {
+            if ($faker->boolean(75)) {
                 $applicantUserSkill->when_skill_used = $faker->randomElement(ApiEnums::whenSkillUsed());
             }
+            // consistent seeding of applicant's skill rankings that matches what is expected
+            // four types independent of each other, starting from 1, with no duplicates or gaps
+            if ($applicantUserSkill->skill->category === 'TECHNICAL') {
+                switch ($applicantUserSkillTechnicalIndex) {
+                    case 0:
+                        $applicantUserSkill->top_skills_rank = 1;
+                        break;
+                    case 1:
+                        $applicantUserSkill->improve_skills_rank = 1;
+                        break;
+                    case 2:
+                        $applicantUserSkill->top_skills_rank = 2;
+                        break;
+                    case 3:
+                        $applicantUserSkill->top_skills_rank = 3;
+                        $applicantUserSkill->improve_skills_rank = 2;
```

Comment:
I'm confused. Is this being marked as both "to improve" and "top skill" at the same time? 😕 I guess it isn't impossible but it seems unlikely that this would represent real data 😅

Reply:
I thought it was possible for a skill you want to both showcase as a top skill and be something you also want to improve further. So intentionally both.

Reply:
Okay, that sort of makes sense. It just seems odd that you would claim a skill as being one of your top skills _**and**_ wanting to improve it. 🤣  Shouldn't we all be striving to always improve all skills we have? 😛 

I guess it seems like we are easily offering users a silly cop out where they just fill up "skills to improve" with all of their "top skills" to make it seem like they have no weaknesses 😕

Reply:
> Shouldn't we all be striving to always improve all skills we have?

Life lessons brought to you by Eric 

> I guess it seems like we are easily offering users a silly cop out where they just fill up "skills to improve" with all of their "top skills" to make it seem like they have no weaknesses

"My weakness is I just can't think of any weaknesses among my endless strengths". 
Might be something to ask, not sure who to ask since that is more a question about the overall big picture